app/src/main/java/org/schabi/newpipe/settings/SelectPlaylistFragment.java

```diff
@@ -0,0 +1,240 @@
+import org.schabi.newpipe.local.playlist.RemotePlaylistManager;
+import org.schabi.newpipe.report.ErrorActivity;
+import org.schabi.newpipe.report.UserAction;
+import org.schabi.newpipe.util.PlaylistItemsUtils;
+
+import java.util.List;
+import java.util.Vector;
+
+import io.reactivex.Flowable;
+import io.reactivex.Observer;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+
+public class SelectPlaylistFragment extends DialogFragment {
+    /**
+     * This contains the base display options for images.
+     */
+    private static final DisplayImageOptions DISPLAY_IMAGE_OPTIONS
+            = new DisplayImageOptions.Builder().cacheInMemory(true).build();
+
+    private final ImageLoader imageLoader = ImageLoader.getInstance();
+
+    private OnSelectedLisener onSelectedLisener = null;
+    private OnCancelListener onCancelListener = null;
+
+    private ProgressBar progressBar;
+    private TextView emptyView;
+    private RecyclerView recyclerView;
+
+    private List<PlaylistLocalItem> playlists = new Vector<>();
+
+    public void setOnSelectedLisener(final OnSelectedLisener listener) {
+        onSelectedLisener = listener;
+    }
+
+    public void setOnCancelListener(final OnCancelListener listener) {
+        onCancelListener = listener;
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Init
+    //////////////////////////////////////////////////////////////////////////*/
+
+    @Override
+    public View onCreateView(@NonNull final LayoutInflater inflater, final ViewGroup container,
+                             final Bundle savedInstanceState) {
+        View v = inflater.inflate(R.layout.select_playlist_fragment, container, false);
+        recyclerView = v.findViewById(R.id.items_list);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+        SelectPlaylistAdapter playlistAdapter = new SelectPlaylistAdapter();
+        recyclerView.setAdapter(playlistAdapter);
+
+        progressBar = v.findViewById(R.id.progressBar);
+        emptyView = v.findViewById(R.id.empty_state_view);
+        progressBar.setVisibility(View.VISIBLE);
+        recyclerView.setVisibility(View.GONE);
+        emptyView.setVisibility(View.GONE);
+
+        final AppDatabase database = NewPipeDatabase.getInstance(this.getContext());
+        LocalPlaylistManager localPlaylistManager = new LocalPlaylistManager(database);
+        RemotePlaylistManager remotePlaylistManager = new RemotePlaylistManager(database);
+
+        Flowable.combineLatest(localPlaylistManager.getPlaylists(),
+                remotePlaylistManager.getPlaylists(), PlaylistItemsUtils::merge)
+                .toObservable()
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(getPlaylistsObserver());
```

Comment:
Just do `.subscribe(this::displayPlaylists, this::onError)` (I didn't test that). You should also store the `Disposable` it returns and dispose it in `onDestroy()`.